File: RetroSpective.Core/Models/Retrospective.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RetroSpective.Core.Models {
    public class RetroSpective {

        public int Id { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public DateTime CreatedDate { get; set; }

        public virtual RetroColumn[] RetroColumns { get; set; }
    }
}

File: Retrospective_Front_End/ClientApp/src/models/Retrospective.ts
import {RetroColumn} from './retroColumn';

export class Retrospective {

  constructor(
    public id: number,
    public title: string,
    public description: string,
    public retroColumns: RetroColumn[]
  ) {
  }

  addRetroColumn(column: RetroColumn) {
    this.retroColumns.push(column);
  }

  removeRetroColumn(columnId: number) {
    this.retroColumns.filter(rc => rc.id !== columnId);
  }
}


